src/org/labkey/test/BaseWebDriverTest.java

```diff
@@ -986,6 +988,20 @@ private void handleFailure(Throwable error, @LoggedParam String testName)
                 // Reset errors before next test and make it easier to view server-side errors that may have happened during the test.
                 checker().withScreenshot(testName + "_serverErrors").wrapAssertion(this::checkErrors);
             }
+
+            // Do last. Heap dump will navigate
+            if (unwrapRuntimeException(error) instanceof DirectoryNotEmptyException notEmpty)
+            {
+                try
+                {
+                    Path path = Paths.get(notEmpty.getFile());
+                    List<String> subdirs = Files.walk(path).map(p -> path.relativize(p).toString())
+                            .filter(s -> !s.isEmpty()).collect(Collectors.toList());
+                    TestLogger.error("Remaining files after attempting to delete: " + path + "\n\t" + String.join("\t\n", subdirs), notEmpty);
+                    getArtifactCollector().dumpHeap();
+                }
+                catch (IOException ignore) { }
```

Comment:
Any value in logging the IOException if it occurs?

Reply:
Now that you mention it, there's not really any reason to not log it.